File: pytools/examples/java/client/protocol/AckRoleRandNameOk.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class AckRoleRandNameOk
{
	private String uname;


	public AckRoleRandNameOk(Packet packet)
	{
		uname = packet.readString();
	}



	public String getUname()
	{
		return this.uname;
	}

}

File: pytools/examples/php/client/protocol/AckRoleRandNameOk.php
<?php
namespace protocol;

use network\Packet;


class AckRoleRandNameOk
{
	private $uname = '';


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$this->uname = $packet->readString();
	}


	public function getUname()
	{
		return $this->uname;
	}

}
